tensorflow/contrib/seq2seq/python/kernel_tests/seq2seq_test.py

```diff
@@ -0,0 +1,114 @@
+                                         input_size])
+        decoder_inputs = tf.constant(0.4, shape=[decoder_sequence_length,
+                                                 batch_size,
+                                                 decoder_input_size])
+        decoder_length = tf.constant(decoder_sequence_length, dtype=tf.int32,
+                                     shape=[batch_size,])
+
+        # Define model
+        encoder_outputs, encoder_state = tf.nn.dynamic_rnn(
+            cell=tf.nn.rnn_cell.GRUCell(encoder_size), inputs=inputs,
+            dtype=tf.float32, time_major=True, scope="rnn")
+        decoder_cell = tf.nn.rnn_cell.GRUCell(decoder_size)
+        decoder_cell = tf.nn.rnn_cell.OutputProjectionWrapper(decoder_cell,
+                                                              output_size)
+
+        # Train decoder
+        decoder_fn_train = tf.contrib.seq2seq.simple_decoder_fn_train(
+            encoder_state=encoder_state)
+        decoder_outputs_train, decoder_state_train = (
+          tf.contrib.seq2seq.dynamic_rnn_decoder(
+              cell=decoder_cell,
+              decoder_fn=decoder_fn_train,
+              inputs=decoder_inputs,
+              sequence_lengths=decoder_length,
+              time_major=True))
+
+        # Inference decoder
+        varscope.reuse_variables()
+        decoder_fn_inference = tf.contrib.seq2seq.simple_decoder_fn_inference(
+            encoder_state=encoder_state,
+            embeddings=decoder_embeddings,
+            start_of_sequence_id=start_of_sequence_id,
+            end_of_sequence_id=end_of_sequence_id,
+            maximum_length=decoder_sequence_length,
+            dtype=tf.int32)
+        decoder_outputs_inference, decoder_state_inference = (
+          tf.contrib.seq2seq.dynamic_rnn_decoder(
+              cell=decoder_cell,
+              decoder_fn=decoder_fn_inference,
+              time_major=True))
+
+        # Run model
+        tf.initialize_all_variables().run()
+        decoder_outputs_train_res, decoder_state_train_res = sess.run(
+            [decoder_outputs_train, decoder_state_train])
+        decoder_outputs_inference_res, decoder_state_inference_res = sess.run(
+            [decoder_outputs_inference, decoder_state_inference])
+
+        # Assert outputs
+        self.assertEqual((decoder_sequence_length, batch_size, output_size),
+                         decoder_outputs_train_res.shape)
+        self.assertEqual((batch_size, decoder_size),
+                         decoder_outputs_inference_res.shape[0:2])
+        self.assertEqual((batch_size, decoder_size),
+                         decoder_state_train_res.shape)
+        self.assertEqual((batch_size, decoder_size),
+                         decoder_state_inference_res.shape)
+        # The dynamic decoder might end earlier than `maximal_length`
+        # under inference
+        true_value = (decoder_sequence_length>=
```

Comment:
@ebrevdo would you like to keep `kernel_tests` as is, or modify `dynamic_rnn_decoder` to also return the sequence_lengths in order to support value assertion? (as elaborated above)
Thanks